packages/zoe/src/contracts/multipoolAutoswap/pool.js

```diff
@@ -0,0 +1,175 @@
+          });
+          return pool.getAmountMath().make(outputValue);
+        }
+        const outputValue = getInputPrice({
+          inputValue,
+          inputReserve: pool.getSecondaryAmount().value,
+          outputReserve: pool.getCentralAmount().value,
+        });
+        return pool.getCentralAmountMath().make(outputValue);
+      },
+      addLiquidity: userSeat => {
+        // Calculate how many liquidity tokens we should be minting.
+        const liquidityValueOut = calcLiqValueToMint(
+          harden({
+            liqTokenSupply,
+            inputValue: userSeat.getAmountAllocated('Central').value,
+            inputReserve: pool.getCentralAmount().value,
+          }),
+        );
+
+        const liquidityAmountOut = liquidityAmountMath.make(liquidityValueOut);
+        liquidityZcfMint.mintGains({ Liquidity: liquidityAmountOut }, poolSeat);
+        liqTokenSupply += liquidityValueOut;
+
+        trade(
+          zcf,
+          {
+            seat: poolSeat,
+            gains: userSeat.getCurrentAllocation(),
+          },
+          {
+            seat: userSeat,
+            gains: { Liquidity: liquidityAmountOut },
+          },
+        );
+        userSeat.exit();
+        return 'Added liquidity.';
+      },
+      removeLiquidity: userSeat => {
+        const liquidityIn = userSeat.getAmountAllocated(
+          'Liquidity',
+          liquidityBrand,
+        );
+        const liquidityValueIn = liquidityIn.value;
+        const centralTokenAmountOut = pool.getCentralAmountMath().make(
+          calcValueToRemove(
+            harden({
+              liqTokenSupply,
+              poolValue: pool.getCentralAmount().value,
+              liquidityValueIn,
+            }),
+          ),
+        );
+
+        const tokenKeywordAmountOut = pool.getAmountMath().make(
+          calcValueToRemove(
+            harden({
+              liqTokenSupply,
+              poolValue: pool.getSecondaryAmount().value,
+              liquidityValueIn,
+            }),
+          ),
+        );
+
+        liqTokenSupply -= liquidityValueIn;
+
+        trade(
+          zcf,
+          {
+            seat: poolSeat,
+            gains: { Liquidity: liquidityIn },
+          },
+          {
+            seat: userSeat,
+            gains: {
+              Central: centralTokenAmountOut,
+              Secondary: tokenKeywordAmountOut,
+            },
+          },
+        );
+
+        userSeat.exit();
+        return 'Liquidity successfully removed.';
+      },
+    };
+    return pool;
+  };
+
+  /**
+   * Allows users to add new liquidity pools. `secondaryIssuer` and
+   * its keyword must not have been already used
+   * @param {Issuer} secondaryIssuer
+   * @param {Keyword} keyword - will be used in the
+   * terms.issuers for the contract, but not used otherwise
+   */
+  const addPool = async (secondaryIssuer, keyword) => {
+    zcf.assertUniqueKeyword(keyword);
```

Comment:
To avoid a game of "guess-the-unused-keyword", can we do something to manage its automatic generation  for the user from something like `Secondary${nonce}`?